subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/protocol/PhasedActionResult.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.internal.protocol;
+
+import java.io.Serializable;
+
+/**
+ * DO NOT CHANGE THIS INTERFACE. It is part of the cross-version protocol.
+ *
+ * A result of one of the actions of an {@link InternalPhasedAction}.
+ * This result will be supplied to {@link PhasedActionResultListener}.
+ *
+ * @since 4.8
+ */
+public interface PhasedActionResult<T> extends InternalProtocolInterface, Serializable {
+
+    /**
+     * Gets the action result if it completed successfully.
+     *
+     * @return The result. {@code null} if a failure occurred.
```

Comment:
❌ This shouldn't happen. If the action failed the build should fail and no result should be returned to the phased handler